canonical-websites/www.canonical.com
# Remove fenchurch

The new build process doesn't support `requirements/server.txt`. All necessary requirements for the app are now in `requirements/standard.txt`.

To facilitate this, we're removing [fenchurch](https://github.com/ubuntudesign/fenchurch), which can't be imported directly from PyPi because it's private.

Instead, we're replacing its functionality with:
- [`django-template-finder-view`](https://github.com/ubuntudesign/django-template-finder-view) - open source module with the automatic template finding functionality from Fenchurch
- A new local view class `CanonicalTemplateFinder` which extends `django_template_finder_view.TemplateFinder` and adds `level_*` context variables (as Fenchurch did).
- [`django-json-redirects`](https://github.com/ubuntudesign/django-json-redirects) - an open-source module replacement for Fenchurch's `redirect.txt` functionality, except now it's in a much nicer JSON format.
# QA

``` bash
make run
```
- Check the site still works as expected.
- Particularly make sure the `active` classes are being added to the navigation markup to highlight the current navigation area - this is what relies on the existence of the `level_*` variables.
- Open `redirects.json` and check some of the redirects still work.
